In [1]:
import pandas as pd
from sklearn import metrics
from sklearn.linear_model import LogisticRegression

In [6]:
cancer = pd.read_csv('knn.csv')

In [7]:
del cancer['id']

#deleting the column, since it doesnot have any purpose/impact with the objective

In [11]:
X = cancer.iloc[:,1:]


In [15]:
y = cancer.iloc[:,0]


In [16]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.2)

In [17]:
model = LogisticRegression()

In [18]:
model.fit(X_train, y_train)

C:\Users\smrithkala\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
          intercept_scaling=1, max_iter=100, multi_class='warn',
          n_jobs=None, penalty='l2', random_state=None, solver='warn',
          tol=0.0001, verbose=0, warm_start=False)

In [19]:
expected = y_test

In [20]:
predicted = model.predict(X_test)

In [21]:
print(metrics.classification_report(expected,predicted))
print(metrics.confusion_matrix(expected,predicted))

              precision    recall  f1-score   support

           B       0.96      0.97      0.97        74
           M       0.95      0.93      0.94        40

   micro avg       0.96      0.96      0.96       114
   macro avg       0.95      0.95      0.95       114
weighted avg       0.96      0.96      0.96       114

[[72  2]
 [ 3 37]]


 We have got a accuracy of 96% (f1-score)
 
 Other Examples of Logistic Regression -  Loan Approval